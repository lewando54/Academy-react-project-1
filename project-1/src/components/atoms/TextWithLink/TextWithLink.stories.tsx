/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import TextWithLink from './TextWithLink'
import { withRouter } from 'storybook-addon-react-router-v6';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: TextWithLink,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    paragraphText: { control: 'text' },
    anchorText: { control: 'text' },
    href: { control: 'text' }
  },

  decorators: [withRouter]
} as ComponentMeta<typeof TextWithLink>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TextWithLink> = (args) => <TextWithLink {...args} />

export const Register = Template.bind({})
export const Login = Template.bind({})

Register.args = {
  paragraphText: 'Already a user?',
  anchorText: 'LOG IN',
  href: '/login'
}

Register.parameters = {
  reactRouter: {
    routePath: '/register'
  }
}

Login.args = {
  paragraphText: 'Dont have an account?',
  anchorText: 'SIGN UP',
  href: '/register'
}

Login.parameters = {
  reactRouter: {
    routePath: '/login'
  }
}
// More on args: https://storybook.js.org/docs/react/writing-stories/args
