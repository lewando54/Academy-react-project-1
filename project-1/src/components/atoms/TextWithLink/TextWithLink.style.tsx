import styled from 'styled-components'

export const StyledDiv = styled.div`
    display: flex;
    gap: 0.25rem/* 4px */;
    font-size: 0.875rem/* 14px */;
    line-height: 1.25rem/* 20px */;
    color: rgb(107 114 128 / 1);
    align-items: center;
`
