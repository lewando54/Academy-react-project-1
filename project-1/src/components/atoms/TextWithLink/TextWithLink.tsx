import React from 'react'
import { useNavigate } from 'react-router-dom'
import { StyledDiv, StyledLink } from './TextWithLink.style'

interface TextWithLinkProps {
  paragraphText: string
  href: string
  anchorText: string
  testId: string
  anchorTagTestId?: string
}

function TextWithLink ({ paragraphText, href, anchorText, testId }: TextWithLinkProps): React.ReactElement {
  const navigate = useNavigate()

  const onClick = (): void => {
    navigate(href)
  }

  return (
        <StyledDiv data-testid={testId}>
            <p>{paragraphText}</p>
            <StyledLink onClick={onClick}>
                {anchorText}
            </StyledLink>
        </StyledDiv>
  )
}

export default TextWithLink
