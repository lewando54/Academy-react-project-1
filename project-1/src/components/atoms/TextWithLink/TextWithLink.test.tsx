import React from 'react'
import '@testing-library/jest-dom'
import * as renderer from 'react-test-renderer'
import TextWithLink from './TextWithLink'
import { fireEvent, render } from '@testing-library/react'
import { MemoryRouter, BrowserRouter, Route, Routes } from 'react-router-dom'

describe('Text With Link', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<MemoryRouter><TextWithLink paragraphText='test' anchorText='click me' href='/' testId='test' anchorTagTestId='atest' /></MemoryRouter>)
    expect(tree).toMatchSnapshot()
  })

  it('should render paragraphText correctly', () => {
    const { getByText } = render(<MemoryRouter><TextWithLink paragraphText='testingP' anchorText='click me' href='/' testId='test' anchorTagTestId='atest' /></MemoryRouter>)
    expect(getByText('testingP')).toBeInTheDocument()
  })

  it('should render anchor tag text correctly', () => {
    const { getByText } = render(<MemoryRouter><TextWithLink paragraphText='testingP' anchorText='click me' href='/' testId='test' anchorTagTestId='atest' /></MemoryRouter>)
    expect(getByText('click me')).toBeInTheDocument()
  })

  it('should navigate in app properly after click', () => {
    const { getByText } = render(
      <BrowserRouter>
        <Routes>
          <Route path='/' element={<TextWithLink paragraphText='testingP' anchorText='click me' href='/test' testId='test' />} />
          <Route path='/test' element={<p>Correct!</p>} />
        </Routes>
      </BrowserRouter>
    )

    fireEvent.click(getByText('click me'))

    console.log(window.location.pathname)

    expect(window.location.pathname).toBe('/test')
  })
})
