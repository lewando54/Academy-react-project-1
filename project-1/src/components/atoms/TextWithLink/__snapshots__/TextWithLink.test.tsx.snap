// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text With Link should render correctly 1`] = `
<div
  className="sc-bcXHqe jFrdJc"
  data-testid="test"
>
  <p>
    test
  </p>
  <p
    className="sc-gswNZR hvpnHU"
    onClick={[Function]}
  >
    click me
  </p>
</div>
`;
